### ISSUE TITLE
Decouple the builder from the sample

### ISSUE THREAD

**Issue by EronWright** [kubernetes-sigs/apiserver-runtime#58]
I observe that the `pkg/builder` uses packages in `internal/sample-apiserver`.  In other words, the library seems hardcoded to the sample.

**Comment by yue9944882** [comment]
@EronWright exactly, and that is designed so that the library will always stick w/ the recommended pattern from the upstream code template. the point is not to "re-implement" an apiserver library, i.e. a light-weight wrapper for making customization from sample-apiserver easier.

**Comment by EronWright** [comment]
I'm sorry, but I still don't understand why the `pkg/builder` would use anything from the sample in `internal/sample-apiserver`, which contains specific API types (`wardle`), a concrete cobra command, and more.  Are you saying that `sample-apiserver` is actually not a sample but is a library?

**Comment by EronWright** [comment]
@yue9944882 please give this issue another look.  Seems there's no way to use apiserver-runtime to make an extension server, because the server is hardcoded to provide the `wardle` type.

**Comment by turkenh** [comment]
> I'm sorry, but I still don't understand why the `pkg/builder` would use anything from the sample in `internal/sample-apiserver`, which contains specific API types (`wardle`), a concrete cobra command, and more. Are you saying that `sample-apiserver` is actually not a sample but is a library?

Any update on this? This is really confusing.

**Comment by Somefive** [comment]
> @yue9944882 please give this issue another look. Seems there's no way to use apiserver-runtime to make an extension server, because the server is hardcoded to provide the `wardle` type.

I also found the use of `internal/sample-apiserver` is a bit confusing. But I think it is possible to use apiserver-runtime to bootstrap an extension server. Like https://github.com/oam-dev/cluster-gateway/blob/master/cmd/apiserver/main.go

Which particular issue are you facing for use it?

**Comment by EronWright** [comment]
I think this is still a valid issue.
/remove-lifecycle rotten
